fhf-datafangst-client/src/components/BenchmarkData/BenchmarkCards.tsx

```diff
@@ -0,0 +1,232 @@
+  const fishingDistance: number[] = [];
+  trips.forEach((trip) => {
+    let fishing: number = 0;
+    trip.hauls.forEach((haul) => {
+      fishing += haul.haulDistance ? haul.haulDistance : 0;
+    });
+    fishingDistance.push(fishing);
+  });
+  return fishingDistance;
+};
+
+const getFishingWeight = (trips: Trip[]) => {
+  const fishingWeight: number[] = [];
+  trips.forEach((trip) => {
+    fishingWeight.push(trip.delivery.totalGrossWeight);
+  });
+  return fishingWeight;
+};
+
+const getTripDates = (trips: Trip[]) => {
+  const dates: string[] = [];
+  trips.forEach((trip) => {
+    dates.push(trip.start);
+  });
+  return dates;
+};
+
+enum BenchmarkType {
+  totalTime,
+  fishingHours,
+  fishingDistance,
+  fishingWeight,
+}
+
+export const BenchmarkCards: FC = () => {
+  const dispatch = useAppDispatch();
+  const trips = useAppSelector(selectTrips);
+  const numHistoric = useAppSelector(selectBenchmarkNumHistoric);
+  if (!trips) {
+    return <></>;
+  }
+
+  const totalTimes = getTotalTimes(trips);
+  const fishingHours = getFishingHours(trips);
+  const fishingDistance = getFishingDistance(trips);
+  const fishingWeight = getFishingWeight(trips);
+  const totalTimeMean =
+    totalTimes.reduce((a, b) => a + b, 0) / totalTimes.length;
+  const fishingHoursMean =
+    fishingHours.reduce((a, b) => a + b, 0) / fishingHours.length;
+  const fishingDistanceMean =
+    fishingDistance.reduce((a, b) => a + b, 0) / fishingDistance.length;
+  const fishingWeightMean =
+    fishingWeight.reduce((a, b) => a + b, 0) / fishingWeight.length;
+
+  const handleClick = (type: BenchmarkType) => {
+    const benchmarkModal: BenchmarkModalParams = {};
+    let data: number[];
+    let metric: string;
+    if (type === BenchmarkType.totalTime) {
+      benchmarkModal.title = "Total tid";
+      benchmarkModal.description =
+        "Total tid er regnet som tiden mellom havneavgang og havneanløp.";
+      metric = "Timer";
+      data = totalTimes;
+    } else if (type === BenchmarkType.fishingHours) {
+      benchmarkModal.title = "Fiske tid";
+      benchmarkModal.description =
+        "Fiske tid er regnet som summen av tiden brukt under hver fangstmelding.";
+      metric = "Timer";
+      data = fishingHours;
+    } else if (type === BenchmarkType.fishingDistance) {
+      benchmarkModal.title = "Fiske distanse";
+      benchmarkModal.description =
+        "Fiske distanse er regnet ut basert på vms/ais meldingene som ble sendt under hver fangstmelding";
+      metric = "Meter";
+      data = fishingDistance;
+    } else if (type === BenchmarkType.fishingWeight) {
+      benchmarkModal.title = "Total vekt";
+      benchmarkModal.description = "Total vekt er basert på total landet vekt";
+      metric = "Kilo";
+      data = fishingWeight;
+    } else {
+      return;
+    }
+    dispatch(setBenchmarkHistoric([metric, getTripDates(trips), data]));
+    dispatch(setBenchmarkModal(benchmarkModal));
+  };
```

Comment:
You can add `metric` and `data` to `BenchmarkModalParams`, then create a
static object that maps `BenchmarkType` to each `BenchmarkModalParams`.

```typescript
const modalParams: Record<BenchmarkType, BenchmarkModalParams> = {
  [BenchmarkType.totalTime]: {
    title: "Total tid",
    description:
      "Total tid er regnet som tiden mellom havneavgang og havneanløp.",
    metric: "Timer",
    data: [],
  },
  ...
};
```

Then this function can be replaced with something like:

```typescript
  const handleClick = (type: BenchmarkType, data: number[]) => {
    const modal = {
      ...modalParams[type],
      data,
    };

    dispatch(setBenchmarkHistoric([modal.metric, getTripDates(trips), data]));
    dispatch(setBenchmarkModal(modal));
  };

```

Reply:
I like how nice the handle click function turned out with this, however the data and the metric is not related to the modal they are related to the chart. Which is why I used two dispatches. I can try to compress handleClick but I think its better to keep the chart and the modal separate.

Reply:
I don't see how they are unrelated, the chart is inside the modal, hence they have a parent/child relation.

If you add data/metric to the modal parameters then you can make the chart component generic by taking in data and metric as properties

```typescript
const modal = useAppSelector(selectBenchmarkModal);
...
<LineChart data={modal.data} metrics={modal.metrics} />
```

```typescript
export const LineChart: FC<Props> = ({ data, metrics }) => {
...
```

Reply:
The thought was that the chart might be used without the modal in the future, but making it generic makes sense. I didn't think of that.

Reply:
Should be resolved as per your wishes